paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -228,6 +231,119 @@ def should_create_piscina_scaling_rule(
     return False, "did not request piscina autoscaling"
 
 
+def should_create_active_requests_scaling_rule(
+    autoscaling_config: AutoscalingParamsDict,
+) -> Tuple[bool, Optional[str]]:
+    """
+    Determines whether we should configure the prometheus adapter for a given service.
+    Returns a 2-tuple of (should_create, reason_to_skip)
+    """
+    if autoscaling_config["metrics_provider"] == "active-requests":
+        return True, None
+    return False, "did not request active-requests autoscaling"
+
+
+def create_instance_active_requests_scaling_rule(
+    service: str,
+    instance: str,
+    autoscaling_config: AutoscalingParamsDict,
+    paasta_cluster: str,
+    namespace: str = "paasta",
+) -> PrometheusAdapterRule:
+    """
+    Creates a Prometheus adapter rule config for a given service instance.
+    """
+    desired_active_requests_per_replica = autoscaling_config[
+        "desired_active_requests_per_replica"
```

Comment:
should there be a default for this? would also maybe make this more ergonomic to use (i.e., a dev wouldn't need to switch to this provider -> run validate -> be told they're missing config)

Reply:
I was thinking about this, but I wasn't sure what a sensible default would be other than 1 😕 We really can't say how many requests a pod can handle. I am sure a pod can handle atleast 1 😃

Reply:
Also just to understand the flow, when will the checks in validate be run? We have added a condition to fail the configuration if the config is missing or less than zero.